SimCalorimetry/EcalTrigPrimAlgos/src/EcalFenixOddAmplitudeFilter.cc

```diff
@@ -0,0 +1,124 @@
+#include "CondFormats/EcalObjects/interface/EcalTPGGroups.h"
+#include "CondFormats/EcalObjects/interface/EcalTPGOddWeightGroup.h"
+#include "CondFormats/EcalObjects/interface/EcalTPGOddWeightIdMap.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include <SimCalorimetry/EcalTrigPrimAlgos/interface/EcalFenixOddAmplitudeFilter.h>
+#include <iostream>
+#include <string>
+#include <fstream>
+
+EcalFenixOddAmplitudeFilter::EcalFenixOddAmplitudeFilter(bool TPinfoPrintout)
+    : inputsAlreadyIn_(0), stripid_{0}, shift_(6), TPinfoPrintout_(TPinfoPrintout) {}
+
+EcalFenixOddAmplitudeFilter::~EcalFenixOddAmplitudeFilter() {}
+
+int EcalFenixOddAmplitudeFilter::setInput(int input) {
+  if (input > 0X3FFFF) {
+    std::cout << "ERROR IN INPUT OF AMPLITUDE FILTER" << std::endl;
+    return -1;
+  }
+  if (inputsAlreadyIn_ < 5) {
+    buffer_[inputsAlreadyIn_] = input;
+    inputsAlreadyIn_++;
+  } else {
+    for (int i = 0; i < 4; i++) {
+      buffer_[i] = buffer_[i + 1];
+    }
+    buffer_[4] = input;
+  }
+  return 1;
+}
+
+void EcalFenixOddAmplitudeFilter::process(std::vector<int> &addout, std::vector<int> &output) {
+  // test
+  inputsAlreadyIn_ = 0;
+  for (unsigned int i = 0; i < 5; i++) {
+    buffer_[i] = 0;  // FIXME: 5
```

Comment:
What about this FIXME?

Reply:
Really old comment inherited from EcalFenixAmplitudeFilter. We can clean up.